### PR TITLE
handle set empty-string (don't use keypather)

### PR DESCRIPTION
**Pull request by tjmehta**

### CODE REVIEW

**Comment by bkendall** [comment]
looks good :) :+1:

**Comment by tjmehta** [comment]
:8ball:

File: lib/RequestData.js [hunk 1]
```diff
@@ -353,9 +358,11 @@ RequestData.prototype.set = function (key, value, formatFunction) {
       });
     }
     else {
-      var upsertVal = (isFunction(formatFunction)) ?
-        formatFunction(keypather.get(req, value) || value) :
-          (keypather.get(req, value) || value);
+      var val = (emptyString(value) || !isString(value)) ?
+        value : (keypather.get(req, value) || value); // keypather edgecase
```
Comment (bkendall) [inline lib/RequestData.js]: this would prevent you from setting `req.body` to `'body'`, but we don't really need that case, I suppose haha

Comment (tjmehta) [inline lib/RequestData.js]: Would it?

**Pull request merged**